ethereumj-core/src/main/java/org/ethereum/casper/core/genesis/CasperStateInit.java

```diff
@@ -0,0 +1,200 @@
+                    ByteUtil.longToBytesNoLeadZeroes(90_000),
+                    deployTx.getSender(),
+                    ByteUtil.bigIntegerToBytes(value),
+                    new byte[0],
+                    null
+            );
+            fundTx.sign(NULL_SIGN_SENDER);
+            txs.add(fundTx);
+            txs.add(deployTx);
+            nonce = nonce.add(BigInteger.ONE);
+        }
+
+        // 0 - fund, 1 - rlp, 2 - fund, 3 - sig hasher, 4 - fund, 5 - purity checker
+        byte[] sigHasherContract = txs.get(3).getContractAddress();
+        byte[] purityCheckerContract = txs.get(5).getContractAddress();
+
+        // Casper!
+        try {
+            // Sources:
+            // https://github.com/ethereum/casper/blob/9106ad647857e6a545f55d7f6193bdc03bb9f5cd/casper/contracts/simple_casper.v.py
+            String casperBinStr = systemProperties.getCasperBin();
+            byte[] casperBin = ByteUtil.hexStringToBytes(casperBinStr);
+
+            CallTransaction.Contract contract = new CallTransaction.Contract(systemProperties.getCasperAbi());
+
+            byte[] casperInit = contract.getConstructor().encodeArguments(
+                    systemProperties.getCasperEpochLength(),  // Epoch length
+                    WITHDRAWAL_DELAY, // Withdrawal delay
+                    ECKey.fromPrivate(sha3("0".getBytes())).getAddress(),  // Owner
+                    sigHasherContract,  // Signature hasher contract
+                    purityCheckerContract,  // Purity checker contract
+                    BASE_INTEREST_FACTOR,  // Base interest factor
+                    BASE_PENALTY_FACTOR,  // Base penalty factor
+                    BigInteger.valueOf(MIN_DEPOSIT_ETH).multiply(BigInteger.TEN.pow(18)) // Minimum validator deposit in wei
+            );
+
+            Transaction tx = new Transaction(
+                    ByteUtil.bigIntegerToBytes(nonce),
+                    ByteUtil.longToBytesNoLeadZeroes(gasPriceFund),
+                    ByteUtil.longToBytesNoLeadZeroes(5_000_000),
+                    new byte[0],
+                    ByteUtil.longToBytesNoLeadZeroes(0),
+                    ArrayUtils.addAll(casperBin, casperInit),  // Merge contract and constructor args
+                    null);
+            tx.sign(NULL_SIGN_SENDER);
+
+            // set casperAddress
+            System.arraycopy(tx.getContractAddress(), 0, casperAddress, 0, 20);
+            txs.add(tx);
+        } catch (Exception ex) {
+            throw new RuntimeException("Failed to generate Casper init transactions", ex);
+        }
+
+        return new Pair<>(casperAddress, txs);
+    }
+
+    private void init() {
+        Repository repo = repository.getSnapshotTo(null);
+        Genesis.populateRepository(repo, genesis);
+
+        // Metropolis dummy
+        repo.saveCode(Hex.decode("0000000000000000000000000000000000000010"), Hex.decode("6000355460205260206020f3"));
```

Comment:
It worth to add a `contracts` section in Genesis file format to keep such specific things there.

Reply:
Originally it was a part of METROPOLIS, I could put it in genesis but not sure that it's the right place for it. Anyway it's a hack so any place is not right :)